lib/python/qmk/info.py

```diff
@@ -9,20 +9,31 @@
 from qmk.constants import ARM_PROCESSORS, AVR_PROCESSORS, VUSB_PROCESSORS
 from qmk.c_parse import find_layouts
 from qmk.keyboard import config_h, rules_mk
+from qmk.makefile import parse_rules_mk_file
 from qmk.math import compute
 
 
 def info_json(keyboard):
     """Generate the info.json data for a specific keyboard.
     """
+    cur_dir = Path('keyboards')
+    rules = parse_rules_mk_file(cur_dir / Path(keyboard) / 'rules.mk')
+    default_folder = rules.get('DEFAULT_FOLDER', '')
+
+    if not default_folder == keyboard:
+        rules = parse_rules_mk_file(cur_dir / Path(default_folder) / 'rules.mk', rules)
+        default_folder = rules.get('DEFAULT_FOLDER', keyboard)
+
+    keyboard = default_folder
```

Comment:
I think this matches the intent behind the make logic and is easier to understand.

```suggestion
    if 'DEFAULT_FOLDER' in rules:
        keyboard = rules.get('DEFAULT_FOLDER') 
        rules = parse_rules_mk_file(cur_dir / keyboard / 'rules.mk', rules)
```

Reply:
No change in error list 👍